helm/doc-ver/values.yaml

```diff
@@ -435,25 +444,163 @@ bundle-visual-anomaly-core-versions:
       model:
         globalStorage:
-            type: gs
-            secret: sa-json
-            bucket: identity-enc-models-prod
+          type: gs
+          secret: sa-json
+          bucket: identity-enc-models-prod
         storage:
-            type: gs
-            secret: sa-json
-            bucket: identity-enc-models-prod
+          type: gs
+          secret: sa-json
+          bucket: identity-enc-models-prod
   models:
     6478fcb410dcce6d3b037199:
       # Model resources can be reduced, however, it is not recommended if
-      # the model is used in production. 
+      # the model is used in production.
       minLimits:
         cpu: 2
         memory: "2Gi"
       maxLimits:
-        cpu: 2
         memory: "2Gi"
       engine:
         type: "triton"
       image:
         repository: "eu.gcr.io/microblink-identity/tritonserver-cpu-onnxruntime/onprem"
-        tag: "23.06"
\ No newline at end of file
+        tag: "23.06"
+
+# this will deploy the opentelemetry collector as a statefulset next to the application
+opentelemetry-collector:
+  mode: statefulset
+  tolerations: []
+
+  config:
+    # -- monitoring tools should be added beforehand and the exporters section should be configured accordingly
+    # -- jaeger/zipkin/prometheus/datadog/...
+    exporters:
+      prometheus:
+        endpoint: 0.0.0.0:8889
+      otlp:
+        endpoint: "release-name-jaeger-collector:4317"
+        tls:
+          insecure: true
+
+    # -- default receivers for the collector, our services support only the grpc receiver
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+            endpoint: 0.0.0.0:4317
+
+    # -- Processors documentation:  https://opentelemetry.io/docs/collector/configuration/#processors
+    processors:
+      batch:
+        send_batch_size: 50000
+        timeout: 3s
+      # -- add custom extra attributes to the traces
+      attributes:
+        actions:
+          key: deployment.environment
+          value: dev
+          action: insert
+      # -- trigger the garbage collector when the memory limit is reached, so that the collector doesn't go out of memory
+      memory_limiter:
+        check_interval: 5s
+        limit_percentage: 80
+        spike_limit_percentage: 50
+      # -- sample 10% of all traces and all traces with status code ERROR/UNSET
+      tail_sampling:
+        # -- time to wait before a decision is made
+        decision_wait: 10s
+        # -- the number of traces to process in the tail sampler
+        num_traces: 100
+        # -- the number of traces you're expected to receive per second
+        # -- it should be modified accordingly to the expected load so that the processor can work better
+        expected_new_traces_per_sec: 10
+        # -- other policies can be added here: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/tailsamplingprocessor
+        policies:
+          - name: trace-10-percent-of-all
```

Comment:
Why is this "tail" sampling? Isn't it just a random decision "before" the request

Reply:
It is a random decision, but it needs to be done after the request is finished since we want to retain all errors also, and if we would do sampling before the request is finished, we wouldn't have that info.
For now, we could just remove tail sampling and use just a probabilistic sampler on the service side, it's easily configurable